src/charm.py

```diff
@@ -19,6 +24,181 @@ class KratosCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._container_name = "kratos"
+        self._container = self.unit.get_container(self._container_name)
+        self._config_file_path = "/etc/config/kratos.yaml"
+        self._identity_schema_file_path = "/etc/config/identity.default.schema.json"
+
+        self.service_patcher = KubernetesServicePatch(self, [("admin", 4434), ("public", 4433)])
+
+        self.database = DatabaseRequires(
+            self,
+            relation_name="pg-database",
+            database_name="database",
+            extra_user_roles="SUPERUSER",
+        )
+
+        self.framework.observe(self.on.kratos_pebble_ready, self._on_pebble_ready)
+        self.framework.observe(self.database.on.database_created, self._on_database_created)
```

Comment:
How about observing `endpoints_changed` as well? They _shouldn't_ change too often, but we could have it just in case